internal/retryhttp/client.go

```diff
@@ -133,3 +175,36 @@ func transport() http.RoundTripper {
 	}
 	return cleanhttp.DefaultPooledTransport()
 }
+
+// retryPolicy is a modified version of retryablehttp.DefaultRetryPolicy to handle
+// status codes differently.
+func retryPolicy(ctx context.Context, resp *http.Response, err error) (bool, error) {
+	// do not retry on context.Canceled or context.DeadlineExceeded
+	if ctx.Err() != nil {
```

Comment:
```suggestion
	if ctx.Err() != nil || err == nil {
```

My bad once again, I was miss-interpreting. This snippet and the rest of the error handling here requires a bit of a double take :p 

Without this 187 can throw a nil pointer panic.

Reply:
The use of the `ok` variable in the type check should avoid a panic https://go.dev/play/p/EE8uZufw6u2

However, the unit tests surfaced a panic with the `nil` response. If the client has exceeded its timeout it will return an error without making the response. It also appears that this error is not a "checkable" `Context deadline exceeded` error so I've added an explicit check for a `nil` response which should cover more bases for us anyways. [This](https://go.dev/play/p/EE8uZufw6u2) is where the error is being returned.